agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryItemPipeline.java

```diff
@@ -0,0 +1,142 @@
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.telemetry;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.io.SerializedString;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.microsoft.applicationinsights.agent.internal.common.OperationLogger;
+import com.microsoft.applicationinsights.agent.internal.exporter.models.TelemetryItem;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.GZIPOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TelemetryItemPipeline {
+
+  private static final Logger logger = LoggerFactory.getLogger(TelemetryItemPipeline.class);
+
+  private static final ObjectMapper mapper = createObjectMapper();
+
+  private static final AppInsightsByteBufferPool byteBufferPool = new AppInsightsByteBufferPool();
+
+  private static final OperationLogger encodeBatchOperationLogger =
+      new OperationLogger(TelemetryItemPipeline.class, "Encoding telemetry batch into json");
+
+  private static ObjectMapper createObjectMapper() {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    // it's important to pass in the "agent class loader" since TelemetryItemPipeline is initialized
+    // lazily and can be initialized via an application thread, in which case the thread context
+    // class loader is used to look up jsr305 module and its not found
+    mapper.registerModules(ObjectMapper.findModules(TelemetryItemPipeline.class.getClassLoader()));
+    mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+    return mapper;
+  }
+
+  private final TelemetryByteBufferPipeline channel;
```

Comment:
rename pipepline since it's no longer channel?

Reply:
I renamed (again)
* `TelemetryByteBufferPipeline` --> `TelemetryPipeline` (I like this shorter name b/c it's the real pipeline and has assocated `TelemetryPipelineRequest/Response/Listener` classes
* `TelemetryItemPipeline` --> `TelemetryItemExporter`